ush/python/pygfs/task/aero_emissions.py

```diff
@@ -37,48 +45,443 @@ def __init__(self, config: Dict[str, Any]) -> None:
+        qfedfiles = [os.path.basename(fname[0]) for fname in config['data_in']['qfed']['copy']]
+        hfedfiles = [os.path.basename(fname[0]) for fname in config['data_in']['hfed']['copy']]
+        gbbepxfiles = [os.path.basename(fname[0]) for fname in config['data_in']['gbbepx']['copy']]
+        climofiles = [os.path.basename(fname[0]) for fname in config['data_in']['climo']['copy']]
+        n_persist = config['n_persist']
 
         localdict = AttrDict(
-            {'variable_used_repeatedly': local_variable}
+            {
+                "cdate": blend_start_date,
+                "nforecast_days": nforecast_hours // 24,
+                # "forecast_dates": forecast_dates,
+                "workdir": self.task_config.DATA,
+                "current_date": self.task_config.PDY,
+                'config': config,
+                'emistype': config['emistype'],
+                'climofiles': climofiles,
+                'qfedfiles': qfedfiles,
+                'hfedfiles': hfedfiles,
+                'gbbepxfiles': gbbepxfiles,
+                'n_persist': n_persist
+            }
         )
 
         # Extend task_config with localdict
         self.task_config = AttrDict(**self.task_config, **localdict)
 
-    @staticmethod
+        # Read the aero_emission.yaml file for common configuration
+
     @logit(logger)
-    def initialize() -> None:
-        """Initialize the work directory
+    def initialize(self) -> None:
+        """Initialize the work directory by copying all the common fix data
+
+        Parameters
+        ----------
+        aero_emission_yaml: Dict
+            Fully resolved aero_emissions.yaml dictionary
+
+         Returns
+        -------
+        None
         """
+        logger.info("Copy Static Data to run directory")
+
+        data_in = self.task_config.config.data_in
+        emistype = self.task_config.emistype
+
+        # Copy climatology files to run directory except for HFED
+        if emistype.lower() != 'hfed':
+            logger.info(
+                f"Copy HFED '{data_in.hfed}' data to run directory"
+            )
+        logger.info(f"Copy climotology data to run directory")
+        FileHandler(data_in.climo).sync()
+        logger.info(f"Copy {emistype} data to run directory")
+        FileHandler(data_in[emistype.lower()]).sync()
+
+    @logit(logger)
+    def run(self) -> None:
+        """
+        Run the AerosolEmissions task with the given parameters.
+
+        Parameters:
+        -----------
+        workdir : Union[str, os.PathLike])
+            The working directory path.
+        current_date: (str, optional)
+            The current date for fire emissions generation.
+        forecast_dates: list, optional
+            List of forecast dates.
+        Config_dict:  Dict
+            Configuration dictionary for the task.
+
+        Returns:
+        --------
+        None
+        """
+        Config_dict = self.task_config['config']
```

Comment:
Object instances should be lowercase.
```suggestion
        config_dict = self.task_config['config']
```

Reply:
This line got changed, but the subsequent references to it weren't.

Reply:
Addressed in 5cdd6a60b